firebase-firestore/src/main/java/com/google/firebase/firestore/remote/Datastore.java

```diff
@@ -215,6 +222,55 @@ public void onClose(Status status) {
     return completionSource.getTask();
   }
 
+  public void runCountQuery(Query query, TaskCompletionSource<Long> result) {
```

Comment:
I'm a bit wary about taking the TaskCompletionSource as an argument, because it's all to easy to forget to set its result, and leave the caller hanging indefinitely. The code looks fine here, but future code changes could easily overlook this. What do you think about changing it back to the style where it returns a Task, the same as this class' commit() method? This would ensure that the Task always gets completed.